e2e/helm.e2e.spec.ts

```diff
@@ -0,0 +1,100 @@
+import path from 'path';
+import {
+  ElectronApplication, BrowserContext, _electron, Page, Locator
+} from 'playwright';
+import { test, expect } from '@playwright/test';
+import { createDefaultSettings, kubectl, helm } from './utils/TestUtils';
+
+let page: Page;
+const defaultReportFolder = path.join(__dirname, 'reports/');
+
+test.describe.serial('Helm Deployment Test', () => {
+  let mainTitle: Locator;
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  const mainTitleSelector = '[data-test="mainTitle"]';
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        '--disable-dev-shm-usage',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: path.join(defaultReportFolder, 'pw-trace.zip') });
+    await electronApp.close();
+  });
+
+  test('should load Rancher Desktop App', async() => {
+    mainTitle = page.locator(mainTitleSelector);
+
+    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+  });
+
+  test('should start loading the background services', async() => {
+    const progressBarSelector = page.locator('.progress');
+
+    // Wait until progress bar show up. It takes roughly ~60s to start in CI
+    await progressBarSelector.waitFor({ state: 'visible', timeout: 200_000 });
+    // Wait until progress bar be detached. With that we can make sure the services were started
+    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
+    await expect(progressBarSelector).toBeHidden();
```

Comment:
CI Only: As we're using Linux runners (Ubuntu 20.04 - Cirrus CI), the `.progress` bar takes a few seconds to be visible and I've implemented an extra `waitFor()` in order to wait the element exists.
With this approach, I haven't found any other flakyness during the k8s api startup in CI.
In parallel, I'm looking for other approaches, but so far I've found only crazy-complex-nonsense solutions... I prefer to keep it simple and human understandable.

Reply:
Does this need to be duplicated to `main.e2e.spec.ts`? Should there be some kind of helper?

Reply:
Yep, sure... good catch. I'll update the main spec.

I'm working on another branch to setup POM (Page Object Model), in order to remove all duplicated code from the existing specs. Still getting couple errors though, the `locator` function cannot detect any element in the Page, when you're invoking it using POM. Probably, I'm missing something and also I've request some guidance from the playwright team. There's no POM implementation for Electron+PW... need to investigate a bit further.